dns/dnssec.py

```diff
@@ -482,6 +482,7 @@ def _sign(
     expiration: Optional[Union[datetime, str, int, float]] = None,
     lifetime: Optional[int] = None,
     verify: bool = False,
+    deterministic_signing: Optional[bool] = None,
```

Comment:
I'm not sure why the default is `None`, rather than `True` or `False`?  Is it to represent an algorithm-specific default if the intention is to have it `True` for ECDSA and `False` for DSA (if it's enabled for ECDSA by default)?

Also, there's no documentation of this new field.

Reply:
The default is `None` as you cannot do deterministic for some algorithms, e.g. DSA for DNSSEC is non-deterministic by default. So my take is that `None` is "do the default for the algorithm" and anything else is explicit choice.

Reply:
Will fix docs.

Reply:
I might be missing something, but I don't think there's a difference between DSA and ECDSA. Neither of them are deterministic by default because the underlying cryptographic algorithm isn't, and RFC 6979 defines deterministic variants of both.

Reply:
It seems to me `cryptography` does not support deterministic signatures for DSA.

Reply:
I can't really blame the `cryptography` maintainers for not wanting to add features to an algorithm that no one should be using anymore,